components/raftstore/src/store/worker/region.rs

```diff
@@ -449,23 +455,29 @@ where
         let overlap_ranges = self
             .pending_delete_ranges
             .drain_overlap_ranges(start_key, end_key);
-        let use_delete_files = false;
-        for (region_id, s_key, e_key) in overlap_ranges {
+        if overlap_ranges.is_empty() {
+            return;
+        }
+        let oldest_sequence = self
+            .engines
+            .kv
+            .get_oldest_snapshot_sequence_number()
+            .unwrap_or(u64::MAX);
```

Comment:
if no snapshot is in use, I think we can safely call `delete_files_in_range`

Reply:
Yes, by setting `oldest_sequence` to `u64::MAX`,  `use_delete_files` always be `true` and  `delete_files_in_range ` will be called.